io/jvm/src/main/scala/fs2/io/net/unixsocket/UnixSocketsPlatform.scala

```diff
@@ -101,12 +102,15 @@ private[unixsocket] trait UnixSocketsCompanionPlatform {
       extends Socket.BufferedReads[F](readMutex) {
 
     def readChunk(buff: ByteBuffer): F[Int] =
-      F.blocking(ch.read(buff)).cancelable(close)
+      evalOnVirtualThreadIfAvailable(F.blocking(ch.read(buff)))
+        .cancelable(close)
 
     def write(bytes: Chunk[Byte]): F[Unit] = {
       def go(buff: ByteBuffer): F[Unit] =
-        F.blocking(ch.write(buff)).cancelable(close) *>
-          F.delay(buff.remaining <= 0).ifM(F.unit, go(buff))
+        evalOnVirtualThreadIfAvailable(
+          F.blocking(ch.write(buff)).cancelable(close) *>
+            F.delay(buff.remaining <= 0).ifM(F.unit, go(buff))
+        )
 
       writeMutex.lock.surround {
         F.delay(bytes.toByteBuffer).flatMap(go)
```

Comment:
Instead of doing the `eval` for every iteration, can just apply it to the first invocation of `go`.